### ISSUE TITLE
Actions created using createAsyncSaga require args

### ISSUE THREAD

**Issue by lawoollett** [franckLdx/create-async-saga#1]
When creating a simple action that dispatches an API request wrapped in `createAsyncSaga` typescript complains if the action has no arguments.

reducer:
```
import { createSlice } from "@reduxjs/toolkit";
import { onFetchTestData } from "@store/sagas/test";

export const DEFAULT_STATE = {
  items: [],
  loading: false,
  error: null,
};

const testSlice = createSlice({
  name: "testReducer",
  initialState: DEFAULT_STATE,
  reducers: {},
  extraReducers(builder) {
    builder.addCase(onFetchTestData.pending, state => {
      state.loading = true;
    });
    builder.addCase(onFetchTestData.fulfilled, (state, action) => {
      state.loading = false;
      state.error = null;
      state.items = [...state.items, ...action.payload];
    });
    builder.addCase(onFetchTestData.rejected, (state, action) => {
      state.loading = false;
      state.error = action.payload.message;
    });
  },
});

export const fetchTestData = onFetchTestData.action;

export const test = testSlice.reducer;
```
saga:
```
import { all, takeEvery } from "redux-saga/effects";
import { createAsyncSaga } from "create-async-saga";
import { getRemoteTestData } from "api";

export const onFetchTestData = createAsyncSaga("test/fetch", function* () {
  const testData = yield getRemoteTestData();
  if (!testData) {
    throw new Error("There was an empty response");
  }
  return testData;
});

export function* testSaga() {
  yield all([takeEvery(onFetchTestData.actionType, onFetchTestData.asyncSaga)]);
}
```

When using the action in a component, e.g. `dispatch(TestActions.fetchTestData());` the following typescript error is shown:
```
Expected 1 arguments, but got 0.ts(2554)
createAction.d.ts(80, 6): Arguments for the rest parameter 'args' were not provided.
const fetchTestData: ActionCreatorWithPreparedPayload
(arg: unknown) => PayloadAction<unknown, string, {
    requestId: string;
}, never>
```

**Comment by franckLdx** [comment]
Hello Luke,

  Thanks for your issue: I had no difficulties to reproduce it.
I'm gonna publish a fix (probably this week-end). Meanwhile, if you want to go on, you can do the following:
`
type FetchDataType = <your data type>
export const onFetchTestData = createAsyncSaga<FetchDataType, void>("test/fetch", function* () {
`

**Comment by franckLdx** [comment]
Resolved in v2.1.10
